scripts/azureml-assets/azureml/assets/environment/pin_image_versions.py

```diff
@@ -11,11 +11,23 @@
 
 from azureml.assets.util import logger
 
+from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_not_exception_type, retry_if_not_exception_message
+from urllib.error import HTTPError
+
 LATEST_TAG = "latest"
 # Handles registry/image_name:{{latest-image-tag}} and registry/image_name:{{latest-image-tag:regex}}
 LATEST_IMAGE_TAG = re.compile(r"([^\"'\s]+):\{\{latest-image-tag(?::(.+))?\}\}")
 
 
+@retry(stop=stop_after_attempt(3), wait=wait_fixed(5), retry=retry_if_not_exception_type(HTTPError) and retry_if_not_exception_message(match=r".*404.*"))

```

Comment:
Would this work instead? Combining conditions should look a little different. https://tenacity.readthedocs.io/en/latest/#whether-to-retry
```suggestion
@retry(stop=stop_after_attempt(3), wait=wait_fixed(5), retry=retry_if_not_exception_message(match=r".*404.*"))
```